gusto/equations.py

```diff
@@ -663,11 +663,15 @@ def __init__(self, domain, parameters, fexpr=None, bexpr=None,
         if fexpr is not None:
             V = FunctionSpace(domain.mesh, "CG", 1)
             f = self.prescribed_fields("coriolis", V).interpolate(fexpr)
-            coriolis_form = coriolis(
-                subject(prognostic(f*inner(domain.perp(u), w)*dx, "u"), self.X))
+            coriolis_form = perp(
+                coriolis(
+                    subject(prognostic(f*inner(u, w)*dx, "u"), self.X)
+                ), domain.perp)
             # Add linearisation
-            linear_coriolis = coriolis(
-                subject(prognostic(f*inner(domain.perp(u_trial), w)*dx, "u"), self.X))
+            linear_coriolis = perp(
```

Comment:
Should we put the linearisation behind an if statement, like for the depth transport term? i.e.
```
if self.linearisation_map(coriolis_form.terms[0]):
   linear_coriolis = ...
   coriolis_form = linearisation(coriolis_form, linear_coriolis)
```
Sorry, I realise we weren't doing this previously!